web/writing.html

```diff
@@ -593,119 +634,30 @@ <h5 class="portlet__head">Webhook</h5>
-    }
-
-    function porkbunCheckedFun() {
-      document.getElementById("DnsID").disabled = false
-      if (beforeDnsID)
-        document.getElementById("DnsID").value = beforeDnsID
-      document.getElementById("dnsIdLabel").innerHTML = "API Key"
-      document.getElementById("dnsSecretLabel").innerHTML = "Secret Key"
-      document.getElementById("dns_help").innerHTML = "<a target='_blank' href='https://porkbun.com/account/api'>创建 Access</a>"
-    }
-
-    function godaddyCheckedFun() {
-      document.getElementById("DnsID").disabled = false
-      if (beforeDnsID)
-        document.getElementById("DnsID").value = beforeDnsID
-
-      document.getElementById("dnsIdLabel").innerHTML = "Key"
-      document.getElementById("dnsSecretLabel").innerHTML = "Secret"
-      document.getElementById("dns_help").innerHTML = "<a target='_blank' href='https://developer.godaddy.com/keys'>创建 API KEY</a>"
-    }
-
-    function googleDomainCheckedFun() {
-      document.getElementById("DnsID").disabled = false
-      if (beforeDnsID)
-        document.getElementById("DnsID").value = beforeDnsID
-      document.getElementById("dnsIdLabel").innerHTML = "Username"
-      document.getElementById("dnsSecretLabel").innerHTML = "Password"
-      document.getElementById("dns_help").innerHTML = "<a target='_blank' href='https://support.google.com/domains/answer/6147083?hl=zh-Hans'>新建动态域名解析记录</a>"
-    }
-
-    function nameCheapCheckedFun() {
-      document.getElementById("dnsIdLabel").innerHTML = ""
-      beforeDnsID = document.getElementById("DnsID").value
-      document.getElementById("DnsID").disabled= true
-      document.getElementById("DnsID").value= ""
-      document.getElementById("dnsSecretLabel").innerHTML = "Password"
-      document.getElementById("dns_help").innerHTML = "<a target='_blank' href='https://www.namecheap.com/support/knowledgebase/article.aspx/36/11/how-do-i-start-using-dynamic-dns/'>如何开启namecheap动态域名解析</a>"
-    }
   </script>
 
   <script>
-    function getLogs(loop = false) {
+    function getLogs() {
       fetch('./logs').then(res => res.text()).then(result => {
-        document.getElementById('logs').innerHTML = result
-      }).finally(() => {
-        if (loop) {
-          setTimeout(getLogs, 5 * 1000, true)
+        const logsDom = document.getElementById('logs')
+        // 判断滚动条是否在底部
+        const isBottom = logsDom.scrollHeight - logsDom.scrollTop - logsDom.clientHeight < 10
+        const preValue = logsDom.value
+        logsDom.value = result
+        // 如果滚动条原先在底部，滚动到底部
+        if (isBottom) {
+          logsDom.scrollTop = logsDom.scrollHeight
+        }
+        // 如果日志有更新且当前未读到，增加未读标记
+        if (preValue !== result && document.querySelector('.logs-panel').style.visibility === 'hidden') {
```

Comment:
```suggestion
        // 如果当前值不为空且日志有更新且当前未读到，增加未读标记
        if (preValue !== "" && preValue !== result && document.querySelector('.logs-panel').style.visibility === 'hidden') {
```

添加判断 `preValue` 是否为空，否则即使日志未更新，每次加载页面时也会添加 `unread`。

Reply:
因为自从上次打开之后也可能出现新的日志，所以这个特判我认为没有必要，除非有强迫症看不得红点

Reply:
我考虑了一下，可以用前端的localStorage来判断是否真的有未读消息

Reply:
或许也可以用 [Server-Sent Events](https://developer.mozilla.org/docs/Web/API/Server-sent_events/Using_server-sent_events)，格式： `data: message\n\n`。不过 `EventSource` 只获取最新的 `data`。

一个使用 Go 的 Server-Sent Events 服务端：https://github.com/mesuutt/ssed

Reply:
刷新显示红点也还好。看修改不

Reply:
没必要为此引入更多的框架

Reply:
不然其实这个日志推送用websoket肯定更好，也更即使